### PR TITLE
Feature/input date

### PR DESCRIPTION
**Pull request by javiercodixis**
**What**

- Feature input date.

**How**

- Using Date picker and AdapterDayjs from mui.

**Screenshots**

image

### CODE REVIEW

File: components/DateInput.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  useFormState,
+  FieldError,
+} from 'react-hook-form';
+import { LocalizationProvider, DatePicker } from '@mui/x-date-pickers';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+
+type DateInputProps = {
+  name: string;
+  control: Control;
+  label?: string; // eslint-disable-line
```
Comment (IrinaCodixis) [inline components/DateInput.tsx]: Why did you disable the eslint rule for these line?

File: components/DateInput.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  ...rest
+}: DateInputProps) => {
+  const { isSubmitted } = useFormState({ control });
+
+  const getErrorMessage = (fieldValue: string, fieldError: FieldError | undefined) => {
+    if (isSubmitted && !fieldValue) {
+      return 'A date must be selected';
+    }
+    switch (fieldError?.type) {
+      case 'maxDate':
```
Comment (IrinaCodixis) [inline components/DateInput.tsx]: If you are not passing maxDate as a rule, do you still get the maxDate error?

File: components/DateInput.tsx (hunk 1 above)
Comment (javiercodixis) [reply components/DateInput.tsx]: If I try to define defaultProps I get this warning
image

Comment (IrinaCodixis) [reply components/DateInput.tsx]: And if you do, what you did in the TextInput (rules = {} - defining the default value on the prop), you still get the error?

Comment (javiercodixis) [reply components/DateInput.tsx]: if I remove this eslint comment 
I get same warning to make defaultProps, which will lead to same warning after the solution of creating defaultProps

image

File: app/page.tsx [hunk 3]
```diff
@@ -135,6 +138,17 @@ const Page = () => {
             rules={{ required: 'You must verify your age to proceed' }}
           />
         </Grid>
+        {isCheckboxChecked
```
Comment (bakycodixis) [inline app/page.tsx]: Prefer this:
```ts
{isCheckboxChecked && (
  // component
)}
```

File: components/DateInput.tsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
+import * as React from 'react';
+import { Controller, Control } from 'react-hook-form';
+import { LocalizationProvider, DatePicker } from '@mui/x-date-pickers';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+
+type DateInputProps = {
```
Comment (bakycodixis) [inline components/DateInput.tsx]: You are not extending `DatePicker` props again.

File: components/DateInput.tsx [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  rules?: object;
+  defaultValue?: string;
+};
+
+const DateInput = ({
+  name,
+  control,
+  label,
+  rules = {},
+  defaultValue = '',
```
Comment (bakycodixis) [inline components/DateInput.tsx]: If you set `defaultValue` like this, then default value for all date input fields in the form will be empty string. We don't want that.

File: components/DateInput.tsx [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+  <LocalizationProvider dateAdapter={AdapterDayjs}>
+    <Controller
+      name={name}
+      control={control}
+      rules={rules}
+      defaultValue={defaultValue}
+      render={({ field, fieldState }) => (
+        <DatePicker
+          label={label}
+          value={field.value || defaultValue}
```
Comment (bakycodixis) [inline components/DateInput.tsx]: Incorrect logic: we should not be defaulting to a value of `defaultValue` if our `field.value` is `null`. If some date is set as default in the form for a given field, but I would like to remove it (leave it empty), I will never be able to do so.

**Review by bakycodixis — CHANGES_REQUESTED** [review]

File: components/DateInput.tsx [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+  <LocalizationProvider dateAdapter={AdapterDayjs}>
+    <Controller
+      name={name}
+      control={control}
+      rules={rules}
+      defaultValue={defaultValue}
+      render={({ field, fieldState }) => (
+        <DatePicker
+          label={label}
+          value={field.value ?? null}
```
Comment (bakycodixis) [inline components/DateInput.tsx:31]: Does `null` value solve your issue for error regarding uncontrolled input turning into controlled?

Comment (javiercodixis) [reply components/DateInput.tsx:31]: Yes, with this fix I dont get any more that error

**Review by bakycodixis — APPROVED** [review]

**Pull request merged**